packages/embark-blockchain-process/src/proxy.js

```diff
@@ -226,6 +235,10 @@ export class Proxy {
       });
     }());
 
+    const web3 = new Web3(`${ws ? 'ws' : 'http'}://${canonicalHost(host)}:${port}`);
+    accounts = (await web3.eth.getAccounts() || []).concat(accounts || []);
+    accounts = [...new Set(accounts.map(ethUtil.toChecksumAddress))];
```

Comment:
Why do you do this? We already do that in the response handling, no?

Reply:
This bootstrapping stuff and is supposed to mirror how the blockchain provider bootstraps the accounts list.

Reply:
Yeah, I understand your idea, but it's not necessary to keep the list of updated accounts, for the moment at least.

`accounts` can stay just the blockchain config accounts (custom accounts using `mnemonic`, etc) and above, we just return the result + `accounts`.

So, in the end, your changes in proxy.js are not really needed, because the proxy already always returned the up to date accounts.

Reply:
Yes, but what it led to was the `eth.getAccounts()` having a different result when comparing web3 usage in the cli dashboard to web3 usage in a browser or a plain Node.js REPL connected to the proxy.

The changes in the provider and the proxy in this PR are intended to keep the `getAccounts()` list up-to-date, consistent, and free of duplicates.

Ideally, I'd like us to discuss and explore doing this operation in just one place, that's why the PR is WIP.

Reply:
Makes sense for the ordering, even though it's not gonna be perfect in terms of matching the array the user puts in `blockchain.js`. I had an old branch where I did some work to order the accounts as per the user's config, but never made a PR because I found that using custom accounts in the Dapp didn't work. You can see the branch here: https://github.com/embark-framework/embark/tree/fix/order-blockchain-accounts

IMO, we can go ahead with your changes as is, since it fixes the problem. We can and will come up with a better solution anyway once we do the refactor, so no need for anything fancy for now.

Reply:
Makes sense, thanks for the link and the feedback.